core/metacling/src/TCling.cxx

```diff
@@ -1265,18 +1265,18 @@ TCling::TCling(const char *name, const char *title, const char* const argv[])
 
    if (fCxxModulesEnabled) {
 #ifdef R__WIN32
-      const char EnvPathDelimiter = ';';
+      constexpr char kEnvPathDelimiter = ';';
 #else
-      const char EnvPathDelimiter = ':';
+      constexpr char kEnvPathDelimiter = ':';
 #endif // R__WIN32
-      auto GetEnvVarPath = [EnvPathDelimiter](const std::string &EnvVar,
-                                              std::vector<std::string> &Paths) {
+      auto GetEnvVarPath = [](const std::string &EnvVar,
```

Comment:
How this capture is unused?

Reply:
because it is a const?

Reply:
>  How this capture is unused?

This is that I get when compiling on macOS:

```
/build/jenkins/workspace/root-pullrequests-build/root/core/metacling/src/TCling.cxx:1272:29: warning: lambda capture 'EnvPathDelimiter' is not required to be captured for this use [-Wunused-lambda-capture]
```

But Windows complains :(

Reply:
That is a bug in the compiler. I disliked all fixes I could come up with. We do not want to capture 'all' -- what about trying to capture explicitly this variable?

Reply:
Did you consider the option of moving the constant outside of the function scope (i.e. a static constexpr in an anonymous namespace)?

Reply:
> We do not want to capture 'all' -- what about trying to capture explicitly this variable?

`[=]` implicitly capture the used automatic variables by copy
see https://en.cppreference.com/w/cpp/language/lambda, section "Lambda capture"
I guess, it is exactly that we need

Reply:
We do not want to copy because of a compiler bug, but that's probably fine.

A more elegant way out is to move this setting as a FoundationUtil inline function returning the correct char (similar to https://github.com/root-project/root/blob/master/core/foundation/res/ROOT/FoundationUtils.hxx#L43-L50). Then the lambda will just use it and will not have to capture anything.

Reply:
Up to you Vassil. Seems to be capture only required on Windows - all other platforms can use *constexpr* as is. Therefore capture with `[=]` does exactly that we need - let compiler decide that it need or not. And value copy in this case may be more efficient then reference.

Reply:
@Axel-Naumann

Reply:
> We do not want to copy because of a compiler bug

Not a compiler bug (constant expressions aren't odr-used), but I don't care at all how we solve it as long as we do. I believe https://github.com/root-project/root/pull/4584 is better simply because it explains what happens in a comment :-) while here, the effect of / need for the `constexpr` here isn't very clear. OTOH here we now have `constexpr` *plus* `[=]` which seems too much, also according to Jenkins's happiness on https://github.com/root-project/root/pull/4584

(Apologies for opening https://github.com/root-project/root/pull/4584 - I overlooked this PR due to vacation...)